### PR TITLE
Updated deprecated calls to fp-ts^2.14.0

### PR DESCRIPTION
**Pull request by parischap**
Updated the parser to use fp-ts^2.14.0
Replaced all deprecated imports
Replaced tslint by eslint and updated scripts accordingly

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -37,21 +39,25 @@
     "fp-ts": "^2.0.0"
```
Comment (gcanti) [inline package.json:39]: peerDependencies should be updated accordingly I guess (`^2.14.0`)

**Review by gcanti — CHANGES_REQUESTED** [review]

**Comment by mjljm** [comment]
The peerDependencies have been updated.
Just for you to know: there remain 3 deprecations in the build.ts script. I had to chose between ApplicativePar and ApplicativeSeq but did not know which one was correct. So preferred to leave things as they were.

**Review by gcanti — APPROVED** [review]

**Comment by gcanti** [comment]
The CI is taking forever, going to merge and then try to update the workflow

**Comment by gcanti** [comment]
Thank you @mjljm, released

**Pull request merged**